packages/formik/src/types.tsx

```diff
@@ -318,6 +339,6 @@ export interface FieldInputProps<Value> {
   onBlur: FormikHandlers['handleBlur'];
 }
 
-export type FieldValidator = (
-  value: any
-) => string | void | Promise<string | void>;
+export type FieldValidatorResponse = string | void | Promise<string | void>;
```

Comment:
Response to me implies there is a Request, like ajax or XHR. I'd go with `FieldValidationResult` as it doesn't imply any kind of Request.